src/libraries/System.Text.Json/tests/System.Text.Json.Tests/SourceGenRegressionTests/Net70RegressionTests.cs

```diff
@@ -0,0 +1,157 @@
+            Assert.Throws<InvalidOperationException>(() => JsonSerializer.Deserialize(json, value.GetType(), Net70GeneratedContext.Default));
+        }
+
+        public static IEnumerable<object[]> GetSupportedTypeRoundtripData_OptionsBased()
+        {
+            yield return Wrap(new ClassWithCustomConverter { Value = 41 }, "42");
+            yield return Wrap(new WeatherForecastWithPOCOs
+            {
+                Date = DateTimeOffset.MinValue,
+                TemperatureCelsius = 10,
+                Summary = "I am a string",
+                DatesAvailable = new List<DateTimeOffset> { DateTimeOffset.MinValue },
+                TemperatureRanges = new Dictionary<string, HighLowTemps>
+                {
+                    ["key"] = new HighLowTemps { Low = 0, High = 5 }
+                },
+                SummaryWords = new[] { "word1", "word2" },
+            },
+            """
+            {
+                "Date" : "0001-01-01T00:00:00+00:00",
+                "TemperatureCelsius" : 10,
+                "Summary" : "I am a string",
+                "DatesAvailable" : [ "0001-01-01T00:00:00+00:00" ],
+                "TemperatureRanges" :
+                {
+                    "key" : { "Low" : 0, "High" : 5 }
+                },
+                "SummaryWords" : [ "word1", "word2" ]
+            }
+            """);
+
+            static object[] Wrap<T>(T value, string expectedJson) => new object[] { value, expectedJson };
+        }
+
+        [Fact]
+        public static void HighLowTemps_ContextReportsCorrectMetadata()
+        {
+            JsonTypeInfo<HighLowTemps> jsonTypeInfo = Net70GeneratedContext.Default.HighLowTemps;
+
+            HighLowTemps instance = jsonTypeInfo.CreateObject();
+            Assert.Equal(0, instance.Low);
+            Assert.Equal(0, instance.High);
+
+            Assert.Equal(2, jsonTypeInfo.Properties.Count);
+
+            JsonPropertyInfo jsonPropertyInfo = jsonTypeInfo.Properties[0];
+            Assert.Equal("High", jsonPropertyInfo.Name);
+            jsonPropertyInfo.Set(instance, 1);
+            Assert.Equal(1, instance.High);
+            Assert.Equal(1, jsonPropertyInfo.Get(instance));
+
+            jsonPropertyInfo = jsonTypeInfo.Properties[1];
+            Assert.Equal("Low", jsonPropertyInfo.Name);
+            jsonPropertyInfo.Set(instance, 2);
+            Assert.Equal(2, instance.Low);
+            Assert.Equal(2, jsonPropertyInfo.Get(instance));
+        }
+
+        [Fact]
+        public static void CombinedContexts_WorksAsExpected()
+        {
+            var options = new JsonSerializerOptions
+            {
+                TypeInfoResolver = JsonTypeInfoResolver.Combine(Net70GeneratedContext.Default, new DefaultJsonTypeInfoResolver())
+            };
+
+            // Unlike v6, v7 Contexts do implement IJsonTypeInfoResolver so combined resolvers will produce the expected output.
```

Comment:
The other tests are identical. Should we have an abstraction?

Reply:
They actually differ in a few places, so I don't think DRY would help a lot here.